### PR TITLE
Hopper

### PR DESCRIPTION
**Pull request by rakeshvel**

### CODE REVIEW

File: 2021-offseason/src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -12,4 +12,19 @@
  * <p>It is advised to statically import this class (or one of its inner classes) wherever the
  * constants are needed, to reduce verbosity.
  */
-public final class Constants {}
+public final class Constants {
+    public static final class kHopper{
```
Comment (KTOmega) [inline 2021-offseason/src/main/java/frc/robot/Constants.java:16]: nit: style: space before opening brace (same for line 25)

File: 2021-offseason/src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -20,8 +23,13 @@
   // The robot's subsystems and commands are defined here...
   private final ExampleSubsystem m_exampleSubsystem = new ExampleSubsystem();
 
+  private final Hopper m_hopper = new Hopper();
+
   private final ExampleCommand m_autoCommand = new ExampleCommand(m_exampleSubsystem);
 
+  XboxController driveController = new XboxController(Constants.kOI.DRIVE_CONTROLLER);
+  XboxController operatorController = new XboxController(Constants.kOI.OPERATOR_CONTROLLER);
```
Comment (KTOmega) [inline 2021-offseason/src/main/java/frc/robot/RobotContainer.java:31]: minor: please put access level on the variable
```suggestion
  private XboxController driveController = new XboxController(Constants.kOI.DRIVE_CONTROLLER);
  private XboxController operatorController = new XboxController(Constants.kOI.OPERATOR_CONTROLLER);
```

File: 2021-offseason/src/main/java/frc/robot/subsystems/Hopper.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import com.ctre.phoenix.*;
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.VictorSPX;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class Hopper extends SubsystemBase {
+  /** Creates a new ExampleSubsystem. */
```
Comment (KTOmega) [inline 2021-offseason/src/main/java/frc/robot/subsystems/Hopper.java:15]: please remove this comment or make it specific to your subsystem

**Review by jaredgoodman03 — APPROVED** [review]
sexy

File: 2021-offseason/src/main/java/frc/robot/subsystems/Hopper.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    right.follow(left);
+    kicker.follow(left);
+
+    left.setInverted(Constants.kHopper.INVERTED);
+    right.setInverted(Constants.kHopper.INVERTED);
+    kicker.setInverted(Constants.kHopper.INVERTED);
+  }
+
+  public void moveForward(){
```
Comment (KTOmega) [inline 2021-offseason/src/main/java/frc/robot/subsystems/Hopper.java:31]: nit: style: space before opening parenthesis (also lines 35, 39)

File: 2021-offseason/src/main/java/frc/robot/subsystems/Hopper.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    left.set(ControlMode.PercentOutput, Constants.kHopper.FORWARD_SPEED);
```
Comment (KTOmega) [inline 2021-offseason/src/main/java/frc/robot/subsystems/Hopper.java:32]: minor: maybe we should abstract out the `left.set(ControlMode.PercentOutput, ...);` to a different method call?

File: 2021-offseason/src/main/java/frc/robot/Constants.java [hunk 6]
```diff
@@ -12,4 +12,19 @@
...
+        public static final int LEFT_ID = 0;
+        public static final int RIGHT_ID = 0;
+        public static final int KICKER_ID = 0;
```
Comment (KTOmega) [inline 2021-offseason/src/main/java/frc/robot/Constants.java:19]: I'm assuming these will be changed once you have the real IDs?

**Review by KTOmega — CHANGES_REQUESTED** [review]
looks good - just had mostly style comments :)

File: 2021-offseason/src/main/java/frc/robot/Constants.java (hunk 6 above)
Comment (jaredgoodman03) [reply 2021-offseason/src/main/java/frc/robot/Constants.java:19]: Yeah we just don't know which motor controllers will be actually used on the robot right now

**Review by KTOmega — APPROVED** [review]
lgtm

**Pull request merged**